### PR TITLE
use dora-sdk-ts

### PR DESCRIPTION
**Pull request by ixje**
I tried to change the code to use `dora-sdk-ts` interfaces instead of having internal ones, but it turned out to be futile. Ultimately components expected data as specific types such as `ParsedTx` and thus we're not gaining anything. At some point it also became unclear to me what the correct place is for data transformations as I sometimes see it happen in components and other times in reducers. I ended up ditching the branch and am now opting to just make sure we're not doing any manual `fetch()`'s and just use the REST interfaces provided by `dora-sdk-ts` so the switch to backend v2 becomes easy.

fwiw; the dora-sdk-ts version update is not strictly necessary, but we'll have to do that at some point and it's useful in for testing with the `/v2/` backend

### CODE REVIEW

**Comment by ixje** [comment]
@comountainclimber I do not get any errors locally, so I'm not sure what's up with circleci

**Comment by comountainclimber** [comment]
Not quite sure what is going on with the type check step but it seems to be finding many type errors 🤔 :
Screenshot 2023-03-21 at 6 55 17 PM

**Comment by comountainclimber** [comment]
Could it be related to the massive diff in `package-lock.json` being introduced with this PR?

File: src/actions/transactionActions.ts [hunk 1]
```diff
@@ -161,44 +150,24 @@ export function fetchTransactions(
...
+          return result.items.map(
+            tx =>
+              ({
+                size: tx.size,
+                time: Number(tx.time),
+                txid: tx.hash,
+                protocol: protocol,
+                network: network,
+              } as Transaction),
+          )
```
Comment (comountainclimber) [inline src/actions/transactionActions.ts:165]: nitpick:
```typescript
  return result.items.map(
             ({time, hash, size}) =>
              ({
                size,
                time: Number(tx.time),
                txid: hash,
                protocol: protocol,
                network: network,
              } as Transaction),
          )
```

File: src/utils/mockData.ts [hunk 2]
```diff
@@ -145,99 +147,15 @@ export const MOCK_BLOCK_LIST_DATA = [
...
-    type: '',
-  },
-]
-
+export const MOCK_TX_LIST_DATA = Array.from(
+  { length: 15 },
+  (_, i) =>
+    ({
+      time: 666,
+      size: 666,
```
Comment (comountainclimber) [inline src/utils/mockData.ts:155]: 😈

**Review by comountainclimber — CHANGES_REQUESTED** [review]
Lets get this type check thing sorted out before merging 😅

**Comment by ixje** [comment]
> Could it be related to the massive diff in `package-lock.json` being introduced with this PR?

No idea why it's there. Even regenerating didn't solve it, but merging the other PR (part 2) and then rebasing this on top of it and rolling back the `package-lock.json` changes finally solved it. Thanks for the hint!

File: src/utils/mockData.ts (hunk 2 above)
Comment (ixje) [reply src/utils/mockData.ts:155]: was already there 😅

File: src/actions/transactionActions.ts (hunk 1 above)
Comment (ixje) [reply src/actions/transactionActions.ts:165]: I did not know you could do that 👍

**Review by comountainclimber — APPROVED** [review]

**Pull request merged**